facebook_connector.py

```diff
@@ -43,34 +44,41 @@ def initialize(dump_directory="."):
         _messages_file = None
 
 
-def resolve_user_id(user_id):
+def resolve_user_id(fb_provided_identifier):
     """
     Tries to map the identifier facebook provides to the name of the user
 
     Args:
-        user_id: identifier string that is provided in the messages file
+        fb_provided_identifier: identifier string that is provided in the messages file
+            This might be the user's name or his Facebook ID
 
     Returns:
         The name of the user if it is able to find it, otherwise the input
     """
-    fb_id_pattern_match = _FB_ID_PATTERN.match(user_id)
-    if fb_id_pattern_match:
-        fb_numeric_id = fb_id_pattern_match.group(1)
-        if fb_numeric_id not in _mapped_fb_ids:
-            username = fb_numeric_id
-            try:
-                fb_user_page = urlopen("https://www.facebook.com/{}".format(fb_numeric_id))
-                fb_page_title = BeautifulSoup(fb_user_page.read()).title.string
-                possible_username = fb_page_title.split("|")[0].strip()
-                if not (username.startswith('Security Check Required') or username.startswith('Page Not Found')):
-                    username = possible_username
-            except:
-                # Wasn't able to find the user - no harm done
-                pass
-            _mapped_fb_ids[fb_numeric_id] = username
+    fb_id_pattern_match = _FB_ID_PATTERN.match(fb_provided_identifier)
+    if not fb_id_pattern_match:
+        # fb_provided_identifier is already the name of the user
+        return fb_provided_identifier
+    fb_numeric_id = fb_id_pattern_match.group(1)
+    if fb_numeric_id in _mapped_fb_ids:
+        # We have looked the id up before, so just return the result
+        return _mapped_fb_ids[fb_numeric_id]
+    try:
+        # Try to find the user id
+        fb_user_page = urlopen("https://www.facebook.com/{}".format(fb_numeric_id))
+        fb_page_title = BeautifulSoup(fb_user_page.read()).title.string
+        possible_username = fb_page_title.split("|")[0].strip()
+        if (possible_username.startswith('Security Check Required') or
+            possible_username.startswith('Page Not Found')):
+            # At some point we probably issue to many request, then we have to give up
+            _mapped_fb_ids[fb_numeric_id] = fb_numeric_id
```

Comment:
On line 73, 76 and 79 (ish since this will change after you make edits) please add something like:

`logging.info('Mapping identifier {0}-->{1}'.format(fb_provided_identifier, possible_username)` or do a similar log statement if an error occurred, e.g. 'Failed to lookup identifier {0}, {1}'.format(fb_provided_identifier, <some minor explanation as to what happened)